mlflow/projects/utils.py

```diff
@@ -102,12 +88,13 @@ def _is_local_uri(uri):
     parsed_uri = urllib.parse.urlparse(uri)
     drive = pathlib.Path(uri).drive
 
+    # Windows local path
     if drive != "" and drive.lower()[0] == parsed_uri.scheme:
-        return not _is_git_repo(uri)
+        return True
     elif parsed_uri.scheme in ("file", ""):
-        return not _is_git_repo(parsed_uri.path)
-    else:
-        return False
+        return True
+
+    return False
```

Comment:
I am wondering if there's a broader purpose to doing the checks this way. 
In version `1.26.1` and before this function was just 
```python
def _is_local_uri(uri):
    return not _GIT_URI_REGEX.match(uri) 
```

and I'm wondering where it broke. Further, wouldn't just this be enough?
```python
def _is_local_uri(uri):
    return Path(uri).exists() and Path(uri).is_dir()
```

Reply:
@ElefHead 

> I'm wondering where it broke

https://github.com/mlflow/mlflow/pull/6000/ might give some context.


> Further, wouldn't just this be enough?

Does `Path(uri).exists()` return True if `uri` is a file URI (e.g. `file:///path/to`)?

Reply:
It would return `False` if the file doesn't exist on local filesystem.

If the function is supposed to, as the docstring says, check whether it should be interpreted as a path on the local filesystem, why not just check whether it is a path on the local filesystem? 

With that interpretation, `Path(uri).exists()` would be enough. The additional check for whether it is a directory is only because a submitted project would technically be a directory. 

Thanks for the links, i'll check it out now -- sorry about the delay, lots of meetings today.